lib/Sema/BuilderTransform.cpp

```diff
@@ -1126,8 +1124,7 @@ ConstraintSystem::matchResultBuilder(AnyFunctionRef fn, Type builderType,
   // not apply the result builder transform if it contained an explicit return.
   // To maintain source compatibility, we still need to check for HasReturnStmt.
   // https://github.com/apple/swift/issues/64332.
-  if (evaluateOrDefault(getASTContext().evaluator,
-                        BraceHasReturnRequest{fn.getBody()}, false)) {
+  if (fn.getBody()->hasExplicitReturnStmt(getASTContext())) {
```

Comment:
One advantage of sticking the member on the AnyFunctionRef is you wouldn't have to pass along the ASTContext, but I don't feel too strongly about it

Reply:
Yeah, that is unfortunate, and storing the context in the `BraceStmt` just for this sounds silly. 

Discoverability was my main concern with declaring it on `AnyFunctionRef`. Would you mind if I add forwarding methods to `AbstractFunctionDecl` and `AbstractClosureExpr` then?

Reply:
Sure, sounds good to me!